### PR TITLE
getLatestReferringParams & getFirstReferringParams exception fix.

### PR DESCRIPTION
**Pull request by starloop-raulbuisan**
Preprocessor directive added to avoid exception when running in the editor.

### CODE REVIEW

**Comment by derrick-branch** [comment]
@raulbuisan thanks for the PR. I'll ask a member of our team to take a look

**Comment by shortstuffsushi** [comment]
Hi @raulbuisan, I wrote this SDK, but Unity is definitely not a strength of mine. Could you explain what these changes are actually doing?

File: Assets/Branch/Branch.cs [hunk 1]
```diff
@@ -68,7 +68,7 @@ public class Branch : MonoBehaviour {
      * Get the referring parameters from the initial install.
      */
     public static Dictionary<string, object> getFirstReferringParams() {
-        string firstReferringParamsString = _getFirstReferringParams();
+        string firstReferringParamsString = Marshal.PtrToStringAnsi(_getFirstReferringParams());
```
Comment (shortstuffsushi) [inline Assets/Branch/Branch.cs:71]: I'm not 100% sure about this, but we typically use UTF8 for everything... Is ANSI going to cause problems?

**Comment by antonargunov** [comment]
Hello, I have reviwed changes, sorry but this code will work only for building iOS version.

For Editor and Android we call method with the next signature:
private static string _getLatestReferringParams()

Marshal.PtrToStringAnsi throws exception because marshalling doesn't work with string type.

For now I have the next code:

public static Dictionary<string, object> getLatestReferringParams() {
        string latestReferringParamsString = "";

```
    #if (UNITY_IOS || UNITY_IPHONE) && !UNITY_EDITOR
    IntPtr ptrResult = _getLatestReferringParams();
    latestReferringParamsString = Marshal.PtrToStringAnsi(ptrResult);
    #else
    latestReferringParamsString = _getLatestReferringParams();
    #endif

    return MiniJSON.Json.Deserialize(latestReferringParamsString) as Dictionary<string, object>;
}
```

Note: I will finish all fixes and changes in Branch in two days

File: Assets/Branch/Branch.cs (hunk 1 above)
Comment (starloop-raulbuisan) [inline Assets/Branch/Branch.cs:71]: Hi,
As far as I tested, I can get custom parameter strings that way. Actually there is no built in method to get those bytes as utf8 string. Probably reading the byte array and then convert it back to a utf8 string could work.

BTW I just made a quick test trying to add some special characters to a parameter and I found out that the create url methods make the app crash too, the url is not created, so I was not able to read back its parameters. I will try to research more deeply this issue.

**Comment by aaustin** [comment]
@raulbuisan Please use the updated SDK - it should resolve this issue.

**Pull request closed without merging**